app/src/org/commcare/CommCareApplication.java

```diff
@@ -729,6 +730,10 @@ public void onServiceConnected(ComponentName className, IBinder service) {
 
                             PurgeStaleArchivedFormsTask.launchPurgeTask();
                         }
+
+                        if (EntityStorageCache.getEntityCacheWipedPref(user.getUsername()) < ReportingUtils.getAppVersion()) {

```

Comment:
Not blocking, but technically I believe we should be using the GUID/internal user ID for this. 

I **think** we also have to require the app ID as part of this key, because technically user storage isn't app isolated right now.